bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Caret.java

```diff
@@ -396,15 +404,7 @@ void restoreIMEFont () {
 public void setBounds (int x, int y, int width, int height) {
 	checkWidget();
 	int zoom = getZoom();
-	setBoundsInPixels(DPIUtil.autoScaleUp(x, zoom), DPIUtil.autoScaleUp(y, zoom), DPIUtil.autoScaleUp(width, zoom), DPIUtil.autoScaleUp(height, zoom));
-}
-
-void setBoundsInPixels (int x, int y, int width, int height) {
-	boolean samePosition = this.x == x && this.y == y;
 	boolean sameExtent = this.width == width && this.height == height;
-	if (samePosition && sameExtent && isCurrentCaret()) return;
```

Comment:
Why is it reasonable to remove this check? Now every time nothing has changed the caret will be resized (running into the else block of the subsequent condition). Is that correct?

Reply:
I think it looks a bit weird on the git diff. It all shows up like that because I changed the parameters to this method (removed width and height). Now that height and width are set in points, we dont need them here. However, if you look in the method down below: https://github.com/eclipse-platform/eclipse.platform.swt/pull/1294/files#r1656807098, the check is still there.

Reply:
So now we set height, width, x and y all in pixels